lib/irb/extend-command.rb

```diff
@@ -255,33 +255,12 @@ def self.def_extend_command(cmd_name, cmd_class, load_file = nil, *aliases)
         cmd_class = cmd_class.name
       end
 
-      if load_file
-        line = __LINE__; eval %[
-          def #{cmd_name}(*opts, **kwargs, &b)
-            Kernel.require_relative "#{load_file}"
-            arity = ::IRB::ExtendCommand::#{cmd_class}.instance_method(:execute).arity
-            args = (1..(arity < 0 ? ~arity : arity)).map {|i| "arg" + i.to_s }
-            args << "*opts, **kwargs" if arity < 0
-            args << "&block"
-            args = args.join(", ")
-            line = __LINE__; eval %[
-              unless singleton_class.class_variable_defined?(:@@#{cmd_name}_)
-                singleton_class.class_variable_set(:@@#{cmd_name}_, true)
-                def self.#{cmd_name}_(\#{args})
-                  ::IRB::ExtendCommand::#{cmd_class}.execute(irb_context, \#{args})
-                end
-              end
-            ], nil, __FILE__, line
-            __send__ :#{cmd_name}_, *opts, **kwargs, &b
-          end
-        ], nil, __FILE__, line
-      else
-        line = __LINE__; eval %[
-          def #{cmd_name}(*opts, &b)
-            ::IRB::ExtendCommand::#{cmd_class}.execute(irb_context, *opts, &b)
-          end
-        ], nil, __FILE__, line
-      end
+      line = __LINE__; eval %[
+        def #{cmd_name}(...)
+          Kernel.require_relative "#{load_file}"
+          ::IRB::ExtendCommand::#{cmd_class}.execute(irb_context, ...)
```

Comment:
This code does not work in ruby < 2.7.2, (test passed because test runs 2.7.8)
https://bugs.ruby-lang.org/issues/16378#note-15
If this is a bug of ruby2.7.0 and 2.7.1, I think we can ignore it. But since this is something like a new feature, I want irb to work with 2.7.0 and 2.7.1. what do you think?

Reply:
I don't think we need to support those patch versions because although it's a new feature, there's no blocker for 2.7.1 users to upgrade to 2.7.2+.

Reply:
But at the same time, it's not a big deal to drop this `...` change. So if you still feel it's better to avoid that, I'm fine with dropping it.

Reply:
I agree that upgrading ruby is not a blocker, but I thought about it and I still feel better to avoid it.
We just dropped 2.6, and this will going to drop 2.7.0 and 2.7.1 so soon. (or maybe we can change the required ruby version in irb.gemspec to 2.7.2)

Reply:
I see, I've dropped that commit 👍

Reply:
Thanks 😄